alphagov/paas-cf-apps-statsd
# [#159174283] Enable the sending of metrics to Loggregator

## What

The paas-metric-exporter has been extended to be able to send metrics to Loggregator. This means it has to write to a local Metron Agent, and therefore is being deployed as a Bosh release.

## How to review

Review and merge https://github.com/alphagov/paas-go/pull/1 first.

Review in conjunction with https://github.com/alphagov/paas-cf/pull/1559. You can deploy it using the branch there. See the notes on that pull request for what we are aiming to achieve.

The main aim of this pull request is for code review. We want to:

* send metrics to Loggregator
* manage a lock for deploying in a highly available manner
* ensure we haven't broken backwards compatibility for anybody using this repository to just send logs to a StatsD endpoint.

## Who

Anyone but me.
